LPS-89180 Allow onChange also when not editing

thektan opened the pull request:
http://issues.liferay.com/browse/LPS-89180

~⚠️There's still a bug where the name doesn't show up until you switch editing on and off.~

Also I'm not sure if https://github.com/andresfulla/liferay-portal/pull/23/commits/f9aa95e95e713609fd395686329aa8f9c06a8e85 breaks anything by removing that check. Not sure why it was needed in the first place yet.

tinatian committed:
LPS-89456 Autogenerated build-upgrade-table

tinatian committed:
LPS-89456 Update to use latest version of LayoutPrototypeTable

tinatian committed:
LPS-89456 Autogenerated build-upgrade-table

tinatian committed:
LPS-89456 Add upgrade process - portal core

tinatian committed:
LPS-89456 Autogenerated buildUpgradeTable - blogs-service

tinatian committed:
LPS-89456 Add upgrade process - blogs-service

tinatian committed:
LPS-89456 Autogenerated buildUpgradeTable - bookmarks-service

tinatian committed:
LPS-89456 Add upgrade process - bookmarks-service

tinatian committed:
LPS-89456 Autogenerated buildUpgradeTable - calendar-service

tinatian committed:
LPS-89456 Add upgrade process - calendar-service

tinatian committed:
LPS-89456 Autogenerated buildUpgradeTable - changeset-service

tinatian committed:
LPS-89456 Add upgrade process - changeset-service

tinatian committed:
LPS-89456 Autogenerated buildUpgradeTable - dynamic-data-lists-service

tinatian committed:
LPS-89456 Add upgrade process - dynamic-data-lists-service

tinatian committed:
LPS-89456 Autogenerated buildUpgradeTable - dynamic-data-mapping-service

tinatian committed:
LPS-89456 Autogenerated buildUpgradeTable - fragment-service

tinatian committed:
LPS-89456 Add upgrade process - fragment-service

tinatian committed:
LPS-89456 Autogenerated buildUpgradeTable - journal-service

tinatian committed:
LPS-89456 Add upgrade process - journal-service

tinatian committed:
LPS-89456 Autogenerated buildUpgradeTable - knowledge-base-service

tinatian committed:
LPS-89456 Add upgrade process - knowledge-service

tinatian committed:
LPS-89456 Autogenerated buildUpgradeTable - layout-page-template-service

tinatian committed:
LPS-89456 Autogenerated buildUpgradeTable - marketplace-service

tinatian committed:
LPS-89456 Add upgrade process - marketplace-service

tinatian committed:
LPS-89456 Autogenerated buildUpgradeTable - message-boards-service

tinatian committed:
LPS-89456 Add upgrade process - message-boards-service

tinatian committed:
LPS-89456 Autogenerated buildUpgradeTable - mobile-device-rules-service

tinatian committed:
LPS-89456 Add upgrade process - mobile-device-rules-service

tinatian committed:
LPS-89456 Autogenerated buildUpgradeTable - oauth2-provider-service

tinatian committed:
LPS-89456 Add upgrade process - oauth2-provider-service

tinatian committed:
LPS-89456 Autogenerated buildUpgradeTable - sync-service

tinatian committed:
LPS-89456 Add upgrade process - sync-service

tinatian committed:
LPS-89456 Autogenerated buildUpgradeTable - wiki-service

tinatian committed:
LPS-89456 Add upgrade process - wiki-service

tinatian committed:
LPS-89456 Autogenerated buildUpgradeTable - portal-workflow-kaleo-service

tinatian committed:
LPS-89456 Add upgrade process - portal-workflow-kaleo-service

tinatian committed:
LPS-89456 Update dependencies on release.schema.version

tinatian committed:
LPS-89456 Add upgrade process - layout-page-template-service

tinatian committed:
LPS-89456 Add support to add extra bad identifiers for each version of portal when building service

tinatian committed:
LPS-89456 Add new reserved/key words added in mysql 8 for 7.2

tinatian committed:
LPS-89456 Remove indexes with bad column names manually, prepare to run build service

tinatian committed:
LPS-89456 Add upgrade process - dynamic-data-mapping-service

tinatian committed:
LPS-89456 Update bad column name in custom sqls

tinatian committed:
LPS-89456 Update bad column names in ***FinderImpl

tinatian committed:
LPS-89456 Baseline

tinatian committed:
LPS-89456 Add BaseUpgradeBadColumnNames

tinatian committed:
LPS-89456 Add support to enable/disable auto update for update sql files, no logic change.

tinatian committed:
LPS-89456 Disable auto update for update sql files in service builder for 7.2

thektan committed:
LPS-89180 Also pass name from selected values

tinatian committed:
LPS-89456 Autogenerated build-upgrade-table - powwow-portlet

tinatian committed:
LPS-89456 Add upgrade process - powwow-portlet

tinatian committed:
LPS-89456 Autogenerated buildUpgradeTable - polls-service

tinatian committed:
LPS-89456 Add upgrade process - polls-service

tinatian committed:
LPS-89456 Ignore integration test, it is testing upgrade process of an old version based on current version of schema, and it fails when current version of schema gets changed.

thektan committed:
LPS-89180 Remove unused props

thektan committed:
LPS-89180 Fetch entity name for id criterion type

thektan committed:
LPS-89180 Allow onChange also when not editing

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

tinatian committed:
LPS-89456 Check if current table is already updated to avoid false warning.

shuyangzhou committed:
LPS-89456 Don't generate useless interface method.

shuyangzhou committed:
LPS-89456 SF

shuyangzhou committed:
LPS-89456 Semantic versioning

thektan committed:
LPS-89180 Add utility function objectToFormData

tinatian committed:
LPS-89456 Make EntityFinder remember where clause using column name in database, prepare for next

tinatian committed:
LPS-89456 Should always use column name in database for sql query

tinatian committed:
LPS-89456 Use the where clause using column name in database for sql query

Commit:
LPS-90233 Adding new change tracking entry resource skeleton

tinatian committed:
LPS-89456 Respect bad column names by invoking getColumnName

Commit:
LPS-90233 Adding CTEntryModel

Commit:
LPS-90233 Adding basic getter service

NemethNorbert committed:
LPS-90239 Don't update editor's button till click action is finished

thektan committed:
LPS-89180 Refactor and fix issues with names updating

tinatian committed:
LPS-89456 Update packageinfo

Commit:
LPS-89547 Lang keys

Commit:
LPS-90144 user-associated-data-web - Create nonrevieable data view

Commit:
LPS-90144 user-associated-data-web - Collapse review data into one step

Commit:
LPS-90144 user-associated-data-web - AnonymizeNonreviewableUADDataMVCRenderCommand.java - methods should be private

Commit:
LPS-90387 Update Clay to v2.8.0

Commit:
LPS-90144 user-associated-data-web - Add results labels

Commit:
LPS-90144 user-associated-data-web - Build lang

Commit:
LPS-90144 Update uad portlet title keys

Commit:
LPS-90144 portal-uad - Build lang

koor4356 committed:
LRQA-45891 Use script instead of java jar for all tests except gogoshel test

Commit:
LPS-89547 Add and update MVC Commands

Commit:
LPS-89547 Update display context for new UI features

Commit:
LPS-89547 Implement UI pieces

Commit:
LPS-89547 Update soy for proper select change list url

Commit:
LPS-89547 build gradle

Commit:
LPS-90233 Fixing error when the sort or limit fields are missing

ealonso committed:
LPS-90286 Creates new management toolbar class

NemethNorbert committed:
LPS-90239 _isClicked should only effect result of showSourceSwitch

NemethNorbert committed:
LPS-90239 Add _onSwitchMouseOut to handle 'mousedown' events without 'click' on _editorSwitch

cgoncas committed:
LPS-90342 Add post operation (Create).

Commit:
LPS-90233 SF

Commit:
LPS-90239 Update npm csf script

ealonso committed:
LPS-89894 Already defined on super class

ealonso committed:
LPS-89894 Creates new management toolbar class

Commit:
LPS-89547 buildLang

ealonso committed:
LPS-89894 Moves common code to display context class

ealonso committed:
LPS-89894 Implements new vertical cards

petershin committed:
LPS-89415 Get test properties from rest-config.yaml

petershin committed:
LPS-89415 Add schemaVarName

petershin committed:
LPS-89415 Add test skeleton templates

cgoncas committed:
LPS-90342 Add put operation (Update).

cgoncas committed:
LPS-90342 Add get operation.

javiergamarra committed:
LPS-90329 fix inconsistent path

Commit:
LPS-90224 Add support for modular conversion of Audience Targeting rules to Segments criteria

Commit:
LPS-90224 Add implementation for browser rule

Commit:
LPS-90224 Add upgrade process for Audience Targeting to Segments

Commit:
LPS-90224 Add test

jbalsas committed:
LPS-89012 Updates dependencies to use liferay-npm-scripts

jbalsas committed:
LPS-90368 Removes unnecessary scripts

Commit:
LPS-90314 users-admin-web - Don't show select button when not part of an org/site

cgoncas committed:
LPS-89658 Add a OData FilterParserProvider

jbalsas committed:
LPS-90387 Updates Clay to v2.8.2

drewbrokke committed:
LPS-90144 user-associated-data-web - AnonymizeNonreviewableUADDataMVCActionCommand - no need to explicityly call redirect

jbalsas committed:
LPS-90387 Adds all-x-items-selected key

drewbrokke committed:
LPS-90144 user-associated-data-web - AnonymizeNonreviewableUADDataMVCRenderCommand - inlines methods that are small and only used once

jbalsas committed:
LPS-90387 Implements Select All and Clear Features

jbalsas committed:
LPS-90387 Enables bulk selection on DocumentLibrary

drewbrokke committed:
LPS-90144 user-associated-data-web - AnonymizeNonreviewableUADDataMVCRenderCommand - iterate over the search container results to get the total count so that we don't have to calculate it twice

cgoncas committed:
LPS-89658 Add .lfrbuild-portal files

drewbrokke committed:
LPS-90144 user-associated-data-web - just use a single constant between the two views

cgoncas committed:
LPS-89658 Add a OData SortParserProvider

cgoncas committed:
LPS-89658 Provide dinamically the EntityModel

brianchiu committed:
LRQA-46715 Use new card macro for assigning site members

drewbrokke committed:
LPS-90144 user-associated-data-web - creates uad_data_navigation_bar.jsp to consolidate nav bar logic

thektan committed:
LPS-89180 Always update state if onChange is called

This is to allow the entity name to be updated in a contributor that isn't currently being edited.

brianchandotcom committed:
LPS-89456 Wordsmith

brianchandotcom committed:
LPS-89456 Wordsmith

brianchandotcom committed:
LPS-89547 SF

brianchandotcom committed:
LPS-89547 Wordsmith

brianchandotcom committed:
LPS-89547 Regen

brianchandotcom committed:
LPS-89547 rename

brianchandotcom committed:
LPS-90387 Wordsmith

brianchandotcom committed:
LPS-90387 Regen

brianchandotcom committed:
LPS-90144 Not needed

cgoncas committed:
LPS-89658 Fix tests.

cgoncas committed:
LPS-89658 Filter values in Structured Content.

javiergamarra committed:
LPS-90229 add support for multipart in vulcan

javiergamarra committed:
LPS-90229 use vulcan multipart instead of cxf

javiergamarra committed:
LPS-90229 expose binary in file

javiergamarra committed:
LPS-90229 remove cxf

brianchandotcom committed:
LPS-90342 Fix typos

javiergamarra committed:
LPS-90229 regenerate

brianchandotcom committed:
LPS-90342 keep it in 1 class

brianchandotcom committed:
LPS-90342 Simplify

brianchandotcom committed:
LPS-90342 Simplify

brianchandotcom committed:
LPS-90342 Rename

brianchandotcom committed:
LPS-90342 SF

brianchandotcom committed:
LPS-90342 SF

brianchandotcom committed:
LPS-90342 SF

brianchandotcom committed:
LPS-90342 let's see how far this pattern goes

brianchandotcom committed:
LPS-90342 Is it possible for a structure to not have a template? I think so?

vicnate5 committed:
LRQA-45891 Add to upgrade suite

thektan committed:
LPS-89180 Add docs to differentiate entityName and modelLabel

tinatian committed:
LPS-89456 Wordsmith

petershin committed:
LPS-90204 Preserve Node.js/NPM versions for 7.1.x

brianchiu committed:
Revert "LPS-89404 Add methods that return a BaseModelSearchResult"

This reverts commit 8735ef11f53ac0c7ca52e9840a59d406354f2034.

brianchiu committed:
Revert "LPS-89404 Add methods that get a list of JournalArticles"

This reverts commit 92d34d4cf0ed93e51f1350271b33d2f0fb8e4ab6.

brianchiu committed:
Revert "LPS-89404 that was just from the generated code"

This reverts commit fc34a829b16a38d6c280e759bd0d2dfb812df115.

brianchiu committed:
Revert "LPS-89404 Add methods that fetch or get a single JournalArticle"

This reverts commit ec06e9d70962a2783956c409116f737a6075b7e8.

brianchiu committed:
Revert "LPS-89404 These methods should also check if the article they would return is being added or updated"

This reverts commit 3aa9c3a7cfc7e71aa6824b1a90554abb36743c49.

brianchiu committed:
Revert "LPS-89404 Avoid NPE"

This reverts commit eab9816900cea30af61ad83baa1cd9f0c135c683.

brianchiu committed:
Revert "LPS-89404 Add simple count methods that don't need additional work (super calls are just avoid breaking the wrapper chain)"

This reverts commit 9f234d79a3569626538dab6dd6059cd6ba1f8118.

brianchiu committed:
Revert "LPS-89404 Update API methods should also set the update in progress flag"

This reverts commit 6e55d84310b19d248e3b711f607eaab018ec26f0.

thektan committed:
LPS-89180 Prevent properties from being dropped into outside contributors

thektan commented:
Finally got around to updating this! 😅 924e5d4...b2f9c04

I also included the prevention of dropping items into the wrong contributors since I removed some of the checks inside `onChange` methods that were previously preventing them.

thektan committed:
LPS-89180 Fix tests

thektan commented:
Realized tests I needed to update the tests, so I pushed up another commit. ✨

brianchandotcom committed:
LPS-90229 Rename

brianchandotcom committed:
LPS-90229 Sort

brianchandotcom committed:
LPS-90229 Move to another package

brianchandotcom committed:
LPS-90229 Simplify

brianchandotcom committed:
LPS-90229 Move everything around now that we have mulitpart

hhuijser committed:
LPS-84119 Do not show status updates by default

brianchandotcom committed:
LPS-90204 prep next

brianchandotcom committed:
Record reference to liferay-portal/master-private with git-commit-portal-private as 838d5e354ee9738dbf241e8a38a20fb4a00e303a

brianchandotcom committed:
LPS-89415 I normally would break, but I think we may add more things in the future

brianchandotcom committed:
LPS-89415 Rename test files

brianchandotcom committed:
LPS-89415 Ignore generated test files

brianchandotcom committed:
LPS-89415 Move to proper package

brianchandotcom committed:
LPS-89415 update build.gradle so tests will at least compile

brianchandotcom committed:
LPS-89415 regen tests

brianchandotcom committed:
LPS-90342 Hugo SF, SF gives me a String out of bounds exception when I try to inline ddmFormValues

tinatian committed:
LPS-89456 Rename properties defined in compatibility.properties

brianchandotcom committed:
LPS-90342 let's consolidate LocalDateTime logic

tinatian committed:
LPS-89456 Apply usages

brianchandotcom committed:
LPS-90342 auto SF

tinatian committed:
LPS-89456 Stop generating useless interface for 7.2 and later, keep the logic unchanged for released versions to avoid package's major version bump

tinatian committed:
LPS-89456 Auto generate

brianchandotcom committed:
LPS-89456 prep next

brianchandotcom committed:
LPS-89456 prep next

brianchandotcom committed:
LPS-89456 auto SF

brianchandotcom committed:
LPS-89456 Undo service.properties changes

brianchandotcom committed:
Record reference to liferay-portal/master-private with git-commit-portal-private as f1d4e9f96f128a18ba721001fdba68ca12835774

ling-alan-huang committed:
LPS-84119 Adjust process order

ling-alan-huang committed:
LPS-84119 Add testcase

brianchandotcom committed:
Record reference to liferay-portal/master-private with git-commit-portal-private as a9592e2d7fea42208023d053f0e6fedf4f6a2380

andresfulla commented:
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

andresfulla commented on modules/apps/segments/segments-web/src/main/resources/META-INF/resources/field/select_organizations.jsp:
```diff
@@ -46,6 +46,15 @@ SelectOrganizationsDisplayContext selectOrganizationsDisplayContext = (SelectOrg
 			keyProperty="organizationId"
 			modelVar="organization"
 		>
+			<%
```
The JS in `select_organizations.jsp` and `select_users.jso` is pretty much the same thing. Couldn't we extract that and leave it as piece that we can use when some other entity needs to be _multiselected_.

thektan replied:
I didn't think it was too concerning at the time, but to keep these js blocks synced I'll create a shared `.jspf` file that both `select_*.jsp` will include.

hhuijser committed:
Revert "LPS-90117 AutoSF, I don't know why this is happening when I execute gradlew formatSource"

This reverts commit ffcc2ad86bcd756ec5fa4b92784eadebd4b01aff.

hhuijser committed:
LPS-90117 Auto SF

andresfulla commented on modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRow.es.js:
```diff
@@ -146,6 +150,61 @@ class CriteriaRow extends Component {
 		supportedPropertyTypes: {}
 	};
 
+	constructor(props, context) {
```
I think grabbing context from the constructor is legacy https://reactjs.org/docs/legacy-context.html, here is the new one https://reactjs.org/docs/context.html#contextconsumer. Not an issue if it works, but it might stop working.

thektan replied:
Nice catch! 🎣Ya it'll be a good idea to update this, so I'll make the changes.

thektan replied:
Actually this was already refactored to call a method `_fetchEntityName` which uses `this.context`. I suppose that's another work around?

Oh and I'm calling it inside of `componentDidMount` instead of the `constructor` too.

andresfulla replied:
Nicely done!!

victorg1991 committed:
LPS-90412 Change application select condition to only attach to the ones that ask for it

victorg1991 committed:
LPS-90412 Remove property not used anymore

ealonso committed:
LPS-89456 Execute buildService on modules

ealonso committed:
LPS-89456 baseline

cgoncas committed:
LPS-90413 Add delete operation.

cgoncas committed:
LPS-90413 Add missing fields.

cgoncas committed:
LPS-90413 Fix type for last reviewed.

Commit:
LRQA-46728 Fix clicking on ellipsis

cgoncas committed:
LPS-90413 Autogenerate.

cgoncas committed:
LPS-90413 Fill field.

cgoncas committed:
LPS-90424 Add fields.

ealonso committed:
LPS-89895 Removes styles

andresfulla commented:
Looks good

javiergamarra committed:
LPS-90429 add missing arrays

javiergamarra committed:
LPS-90429 parse properties

javiergamarra committed:
LPS-90429 use properties

patriciajeanperez committed:
LRQA-46704 Use ExpandInfo macro to check if info panel is open

brianchandotcom committed:
LPS-84119 prep next

brianchandotcom committed:
LPS-90424 Simplify

brianchandotcom committed:
LPS-90424 Sort

brianchandotcom committed:
LPS-89658 Sort

brianchandotcom committed:
LPS-89658 We can just call this _serviceRegistration, since we only have one. MockResource is called mockResource because it IS a MockResource. ServiceRegistration can be called serviceRegistration because the class it IS is not a MockServiceRegistration

brianchandotcom committed:
LPS-89658 Sort

brianchandotcom committed:
LPS-89658 I don't like using the name *Helper unless it's a class that does many things. In this case, it provides a list of entity fields

brianchandotcom committed:
LPS-89658 Keep things simple. We're in a private method. There's no need for Optional because no one is going to accidentally mess up its usage, since we're the only ones using this

brianchandotcom committed:
LPS-89658 Rename

brianchandotcom committed:
LPS-89658 Account for both methods

brianchandotcom committed:
LPS-89658 Account for all 3 methods

brianchandotcom committed:
LPS-89658 Sort

vicnate5 committed:
LRQA-45396 Enable tests

jbalsas committed:
LPS-90387 Updates Clay to v2.8.3

jbalsas committed:
LPS-90387 Fixes ManagementToolbar tests

petershin committed:
LPS-89033 Move logic in RESTBuilder

brianchandotcom committed:
LPS-89658 The special hidden static method is no longer needed

petershin committed:
LPS-90429 Generated

jpince committed:
LPS-89547 Auto SF

jpince committed:
LPS-90351 Auto SF

brianchandotcom committed:
LPS-89658 This should be private since no one else uses it

jpince committed:
LPS-90144 Auto SF - dumb quotes

tinatian committed:
LPS-89456 Auto generate for modules

jpince committed:
LPS-89547 SF

victorg1991 committed:
LPS-90442 Implement get for ddmstructure

tinatian committed:
LPS-89456 Baseline

brianchandotcom committed:
LPS-89658 auto SF

brianchandotcom committed:
LPS-89658 Use an interface

brianchandotcom committed:
LPS-89658 Class can never be null?

brianchandotcom committed:
LPS-89658 Prevet Long parse error on null string

brianchandotcom committed:
LPS-89658 rename

brianchandotcom committed:
LPS-89658 auto SF

brianchandotcom committed:
LPS-89547 like this

brianchandotcom committed:
LPS-90442 Simplify

brianchandotcom committed:
LPS-90429 regen

thektan committed:
LPS-89180 Combine repeated logic

This also fixes an issue where the select all checkbox was not selecting any rows.

thektan commented:
Sent to Bryce https://github.com/bryceosterhaus/liferay-portal/pull/25 

My change: e063ea2
